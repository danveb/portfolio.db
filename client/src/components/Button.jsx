import { Link } from "react-router-dom";
import rightArrow from "../assets/button/right-arrow.png"; 
import rightArrowAlt from "../assets/button/right-arrow-alt.png"; 
import "../styles/Button.css";

export default function Button({ title, url, rel=null, target=null }) {
  return (
    <div className="button__container">
      <Link to={url} className={url.indexOf("https://github.com") !== -1 ? "button--cta--alt" : "button--cta"} rel={rel} target={target}>
        {title}
        <img src={url.indexOf("https://github.com") !== -1 ? rightArrowAlt : rightArrow} alt="right arrow"/>
      </Link>
    </div>
  ) 
}