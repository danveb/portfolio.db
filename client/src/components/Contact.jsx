import "../styles/Contact.scss"; 

const Contact = () => {  
    return (
        <div id="contact" className="contact">
            <div className="contact__wrapper">
                <div className="contact__main">
                    <div className="contact__title">
                        <h1>Send Me A Note<span>.</span></h1>
                    </div>
                    <div className="contact__intro">
                        <p>I'm open to discussing new opportunities to work together. Feel free to send a note and I'll do my best to get back to you!</p>
                        <a href="mailto:danibae91@gmail.com?subject=👨‍💻  Hi Danny Let's Work Together!" target="_blank" rel="noreferrer noopener" className="cta">Say Hello</a>
                    </div>
                </div>
            </div>
        </div>
    )
}

export default Contact