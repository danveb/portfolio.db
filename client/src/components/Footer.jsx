import { Link } from "react-router-dom";
import { socials } from "../constants/socials";
import "../styles/Footer.scss"; 

export default function Footer() {
    return (
        <div className="footer">
            <div className="footer__wrapper">
                <div className="footer__main">
                    {/* <div className="footer__head">
                        <h1>Social Links<span>.</span></h1>
                    </div> */}
                    <div className="footer__copyright">
                        <p>© All rights reserved – Danny Bae</p>
                    </div>
                    <div className="footer__social__icons">
                        <ul>
                            {socials.map((social) => (
                                <li key={social.id}>
                                    <Link to={social.url} target={social.target} rel={social.rel}>
                                        <img src={social.img} alt={social.alt} />
                                    </Link>
                                </li>
                            ))}
                        </ul>
                    </div>
                </div>
            </div>
        </div>
    )
}
