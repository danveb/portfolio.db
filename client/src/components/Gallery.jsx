import asador from "../assets/gallery/asador_blank.png"; 
import bobaLife from "../assets/gallery/bobaLife_blank.png"; 
import sailly from "../assets/gallery/sailly_blank.png"; 
import ioniq5 from "../assets/gallery/ioniq5_blank.png"; 
import coini$t from "../assets/gallery/coini$t_blank.png"; 
import "../styles/Gallery.css"; 

export default function Gallery() {
  return (
    <div className="gallery">
      <div className="gallery__wrapper">
        <div className="gallery__card">
          <img src={sailly} alt="" />
          <img src={asador} alt="" />
          <img src={bobaLife} alt="" />
          <img src={ioniq5} alt="" />
          <img src={coini$t} alt="" />
        </div>
      </div>
    </div>
  )
}