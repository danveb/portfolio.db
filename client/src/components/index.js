export { default as Navbar } from "./Navbar"; 
export { default as Menu } from "./Menu"; 
export { default as Gallery } from "./Gallery"; 
export { default as Contact } from "./Contact"; 
export { default as Footer } from "./Footer"; 
export { default as ProjectList } from "./ProjectList"; 
export { default as ProjectDetail } from "./ProjectDetail"; 