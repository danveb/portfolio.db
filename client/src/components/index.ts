export { default as Navbar } from "./Navbar"; 
export { default as Menu } from "./Menu"; 
export { default as Button } from "./Button"; 
export { default as Gallery } from "./Gallery"; 
export { default as Footer } from "./Footer"; 
export { default as ProjectList } from "./ProjectList"; 
export { default as ProjectDetail } from "./ProjectDetail"; 
export { default as ScrollToTop } from "./ScrollToTop"; 
export { default as MapboxGL } from "./MapboxGL"; 
export { default as Note } from "./Note"; 
export { default as MessageSent } from "./MessageSent"; 