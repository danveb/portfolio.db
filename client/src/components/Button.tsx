import { Link } from "react-router-dom";
import rightArrow from "../assets/button/right-arrow.png"; 
import rightArrowAlt from "../assets/button/right-arrow-alt.png"; 
import "../styles/Button.css";
import { ButtonProps } from "../types";

export default function Button({ title, url, rel, target }: ButtonProps) {
  return (
    <div className="button__container" data-testid="button">
      <Link to={url} className={url.indexOf("https://github.com") !== -1 ? "button--cta--alt" : "button--cta"} rel={rel} target={target}>
        {title}
        <img src={url.indexOf("https://github.com") !== -1 ? rightArrowAlt : rightArrow} alt="right arrow"/>
      </Link>
    </div>
  ) 
}