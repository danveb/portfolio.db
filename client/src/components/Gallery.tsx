import { projects } from "../constants/projects";
import "../styles/Gallery.css";

export default function Gallery() {
  return (
    <div className="gallery">
      <div className="gallery__wrapper">
        <div className="gallery__card">
          {projects.map((project) => (
            <img src={project.img} alt={project.alt} />
          ))}
        </div>
      </div>
    </div>
  )
}