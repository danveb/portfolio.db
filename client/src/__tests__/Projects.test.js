import { render, screen } from "@testing-library/react"; 
import { Projects } from "../components";

describe("Projects component", () => {
    // smoke test
    it("renders without crashing", () => {
        render(<Projects />); 
    }); 

    // snapshot test
    test("matches snapshot", () => {
        const { asFragment } = render(<Projects />); 
        expect(asFragment()).toMatchSnapshot(); 
    });

    // getByText
    test("should display the correct title of bobaLife project", () => {
        render(<Projects />); 
        const title = screen.getByText("bobaLife"); 
        expect(title).not.toBe("bobalife"); 
    });

    // getByText
    test("should read a piece of description of bobaLife project", () => {
        render(<Projects />); 
        const description = screen.getByText("bobaLife is an e-commerce web app", { exact: false }); 
        expect(description).toBeInTheDocument(); 
    });

    // getByRole 
    test("should display more than 20 tags", () => {
        render(<Projects />); 
        const tags = screen.getAllByRole("button"); 
        expect(tags.length).toBeGreaterThan(20); 
    });

    // getElementsByClassName
    test("project cards should display 5 external links to GitHub", () => {
        render(<Projects />); 
        const projects = screen.getAllByRole("link", { name: /GitHub Repo/i }); 
        expect(projects.length).toBeGreaterThan(4); 
    }); 
}); 