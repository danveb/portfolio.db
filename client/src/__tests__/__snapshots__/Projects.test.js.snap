// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Projects component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="projects"
    id="projects"
  >
    <div
      class="projects__wrapper"
    >
      <div
        class="projects__main"
      >
        <div
          class="projects__title"
        >
          <h3>
            Projects
            <span>
              .
            </span>
          </h3>
        </div>
        <div
          class="projects__cards"
        >
          <div
            class="card__container"
          >
            <div
              class="image__container"
            >
              <img
                alt="portfolio project showcase"
                src="sail-ly.png"
              />
            </div>
            <div
              class="card__content"
            >
              <div
                class="card__title"
              >
                <h1>
                  sail.ly
                </h1>
              </div>
              <div
                class="card__body"
              >
                <p>
                  Dream in open waters. SAIL.LY is a web app for beginner-to-experienced sailors that want to enjoy a day out on the water. App allows users to check on their favorite sailing clubs for sailing conditions based on real-time weather data, integrated with Weather API.
                </p>
              </div>
              <div
                class="btn"
              >
                <button>
                  React
                </button>
              </div>
              <div
                class="btn"
              >
                <button>
                  CSS
                </button>
              </div>
              <div
                class="btn"
              >
                <button>
                  Node
                </button>
              </div>
              <div
                class="btn"
              >
                <button>
                  Express
                </button>
              </div>
              <div
                class="btn"
              >
                <button>
                  PostgreSQL
                </button>
              </div>
              <div
                class="btn"
              >
                <button>
                  Weather API
                </button>
              </div>
              <div
                class="btn"
              >
                <button>
                  MapboxGL API
                </button>
              </div>
              <div
                class="btn"
              >
                <button>
                  Netlify
                </button>
              </div>
              <div
                class="card__footer"
              >
                <a
                  href="https://sail-ly.netlify.app"
                  rel="noreferrer noopener"
                  target="_blank"
                >
                  Live Site
                </a>
                <a
                  href="https://github.com/danveb/sail.ly"
                  rel="noreferrer noopener"
                  target="_blank"
                >
                  GitHub Repo
                </a>
              </div>
            </div>
          </div>
          <div
            class="card__container"
          >
            <div
              class="image__container"
            >
              <img
                alt="portfolio project showcase"
                src="bobaLife.png"
              />
            </div>
            <div
              class="card__content"
            >
              <div
                class="card__title"
              >
                <h1>
                  bobaLife
                </h1>
              </div>
              <div
                class="card__body"
              >
                <p>
                  Who enjoys sipping an ice-cold boba milk tea at any time of the day? bobaLife is an e-commerce web app designed for a fictitous boba shop. This app incorporates working cart/checkout process and integration of Stripe mobile payments.
                </p>
              </div>
              <div
                class="btn"
              >
                <button>
                  React
                </button>
              </div>
              <div
                class="btn"
              >
                <button>
                  Redux
                </button>
              </div>
              <div
                class="btn"
              >
                <button>
                  Styled Components
                </button>
              </div>
              <div
                class="btn"
              >
                <button>
                  Express
                </button>
              </div>
              <div
                class="btn"
              >
                <button>
                  JWT
                </button>
              </div>
              <div
                class="btn"
              >
                <button>
                  Stripe-js
                </button>
              </div>
              <div
                class="btn"
              >
                <button>
                  MongoDB
                </button>
              </div>
              <div
                class="btn"
              >
                <button>
                  Netlify
                </button>
              </div>
              <div
                class="card__footer"
              >
                <a
                  href="https://bobalife.netlify.app"
                  rel="noreferrer noopener"
                  target="_blank"
                >
                  Live Site
                </a>
                <a
                  href="https://github.com/danveb/bobalife"
                  rel="noreferrer noopener"
                  target="_blank"
                >
                  GitHub Repo
                </a>
              </div>
            </div>
          </div>
          <div
            class="card__container"
          >
            <div
              class="image__container"
            >
              <img
                alt="portfolio project showcase"
                src="asador.png"
              />
            </div>
            <div
              class="card__content"
            >
              <div
                class="card__title"
              >
                <h1>
                  asador
                </h1>
              </div>
              <div
                class="card__body"
              >
                <p>
                  Ever wondered about asado (barbeque) in Buenos Aires? Asador is all about sharing the joy of the best parrillas (steakhouses). Users are greeted with a live map with pins. Signing up for an account allows users to pin their favorite parrillas.
                </p>
              </div>
              <div
                class="btn"
              >
                <button>
                  React
                </button>
              </div>
              <div
                class="btn"
              >
                <button>
                  Redux
                </button>
              </div>
              <div
                class="btn"
              >
                <button>
                  Sass
                </button>
              </div>
              <div
                class="btn"
              >
                <button>
                  Express
                </button>
              </div>
              <div
                class="btn"
              >
                <button>
                  JWT
                </button>
              </div>
              <div
                class="btn"
              >
                <button>
                  MapboxGL API
                </button>
              </div>
              <div
                class="btn"
              >
                <button>
                  MongoDB
                </button>
              </div>
              <div
                class="btn"
              >
                <button>
                  Netlify
                </button>
              </div>
              <div
                class="card__footer"
              >
                <a
                  href="https://asador.netlify.app"
                  rel="noreferrer noopener"
                  target="_blank"
                >
                  Live Site
                </a>
                <a
                  href="https://github.com/danveb/asador.ly"
                  rel="noreferrer noopener"
                  target="_blank"
                >
                  GitHub Repo
                </a>
              </div>
            </div>
          </div>
          <div
            class="card__container"
          >
            <div
              class="image__container"
            >
              <img
                alt="portfolio project showcase"
                src="ioniq5.png"
              />
            </div>
            <div
              class="card__content"
            >
              <div
                class="card__title"
              >
                <h1>
                  ioniq5 Lookup
                </h1>
              </div>
              <div
                class="card__body"
              >
                <p>
                  App designed for users who want to search deliveries of 2022 Hyundai IONIQ 5 to local dealerships. User can fill in the zipcode and radius for searching. Data is fetched from Hyundai USA's API and key details are displayed on a table.
                </p>
              </div>
              <div
                class="btn"
              >
                <button>
                  React
                </button>
              </div>
              <div
                class="btn"
              >
                <button>
                  Sass
                </button>
              </div>
              <div
                class="btn"
              >
                <button>
                  MUI
                </button>
              </div>
              <div
                class="btn"
              >
                <button>
                  Express
                </button>
              </div>
              <div
                class="btn"
              >
                <button>
                  Hyundai USA API
                </button>
              </div>
              <div
                class="btn"
              >
                <button>
                  Netlify
                </button>
              </div>
              <div
                class="card__footer"
              >
                <a
                  href="https://ioniq5-lookup.netlify.app/"
                  rel="noreferrer noopener"
                  target="_blank"
                >
                  Live Site
                </a>
                <a
                  href="https://github.com/danveb/ioniq5"
                  rel="noreferrer noopener"
                  target="_blank"
                >
                  GitHub Repo
                </a>
              </div>
            </div>
          </div>
          <div
            class="card__container"
          >
            <div
              class="image__container"
            >
              <img
                alt="portfolio project showcase"
                src="laptop1.png"
              />
            </div>
            <div
              class="card__content"
            >
              <div
                class="card__title"
              >
                <h1>
                  Check my GitHub
                </h1>
              </div>
              <div
                class="card__body"
              >
                <p>
                  Check my GitHub link to view more projects where I continue to build user-focused apps. Follow the link below!
                </p>
              </div>
              <div
                class="card__footer"
              >
                <a
                  href=""
                  rel="noreferrer noopener"
                  target="_blank"
                />
                <a
                  href="https://github.com/danveb"
                  rel="noreferrer noopener"
                  target="_blank"
                >
                  GitHub Repo
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
